File: 45-jump-game-ii/jump-game-ii.py
class Solution(object):
    def jump(self, nums):
        n = len(nums)
        jump = 0
        curr = 0
        far = 0
        for i in range(n-1):
            far = max(far,i+nums[i])
            if(curr==i):
                jump += 1
                curr = far
        return jump
        

File: 45-jump-game-ii/jump-game-ii.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var jump = function(nums) {
    let n = nums.length;
    let jump = 0;
    let far = 0;
    let curr = 0;
    for(let i=0;i<n-1;i++){
        far = Math.max(far,i+nums[i]);
        if(curr==i){
            jump++;
            curr = far;
        }
    }
    return jump;
};